src/lib.rs

```diff
@@ -569,3 +652,36 @@ cfg_if! {
         }
     }
 }
+
+#[cfg(any(
+    target_os = "windows",
+    target_os = "illumos",
+    target_os = "solaris",
+    target_os = "vxworks",
+    target_os = "fuchsia",
+    target_os = "horizon",
+    all(
+        unix,
+        not(any(
+            target_os = "linux",
+            target_os = "android",
+            target_os = "macos",
+            target_os = "ios",
+            target_os = "tvos",
+            target_os = "watchos",
+            target_os = "freebsd",
+            target_os = "netbsd",
+            target_os = "openbsd",
+            target_os = "dragonfly",
+        ))
+    )
+))]
```

Comment:
I don't think this overly large `cfg` restriction makes much sense, just put an `allow(unused)` here.